### DISCUSSION TITLE
Get current page margins

### DISCUSSION THREAD

**Discussion by qooxzuub** [typst/typst#3022] [Q&A] [answered]
How can I obtain the current page margins? I found documentation explaining how to set the margins, but not how to query them.

I want to test whether the current location is at the left-hand margin, which I could do with ```locate``` if I knew how to get the left-hand margin as a float (in mm, say).

(The motivation is to build a better QED marker at the end of proofs: the implementations I have seen just add ```h(1fr)``` and then the box, and the box can end up on the left of the page if the last line of the proof fills a whole line.)

**Comment by Dherse** [comment, marked as answer]
I think your best guess for now is to either use layout, use the location and the page number, or wait for get rules. It is planned to allow you to query set rules on your doc. But it’s not there yet. 😥 

but it will come and then it’ll be very easy 😁

**Comment by qooxzuub** [comment]
Thanks! I don't see how to do it with layout + location + page so I guess I'll just wait for the getters to be implemented :-)

**Reply by Dherse** [reply]
Essentially, you can wrap all of your pages in a layout to get the size, but it's really ugly. I think a nicer way would be to use a locate block to get the current location from which you can query the absolute page number (irrespective of the page counter). This means that you can check whether you're an even or odd page and obtain the margin yourself (using a global variable or just hard coding it). It's not ideal but it should work decently

**Comment by Complex2-Liu** [comment]
What you really need is the `sym.wj`

```typst
#let proof(body, spacing: 1.75em, name: "Proof") = {
  let c = {
    emph(name + ".") + body + h(1fr) + sym.wj + $square.stroked$
  }
  block(above: spacing, below: spacing, width: 100%, c)
}

#proof[
  Trivial. Left as an exercise :).
]
```
This prevents the QED symbol from appearing on the left side of the page if the last line of the proof fills a full line.
